### PR TITLE
fit_regularized for WLS/GLS (closes #3559)

### LINKED ISSUE #3559: statsmodels 0.8.0 is missing smf.WLS.fit_regularized
I have been using statsmodels 0.6.0 and was able to train models using
```
import statsmodels.api as smf
smf.WLS.fit_regularized()
```
This allowed me to fit a weighted least squares estimation with L1/L2 regularization.
I recently upgraded to statsmodels 0.8.0 to access improved time series models, but do not see the WLS regularized module anymore. 

In statsmodels 0.8.0 we have:
```
import statsmodels.api as smf
smf.OLS.fit_regularized()
```
which is not the same thing and missing the weight parameters. How can i apply smf.WLS.fit_regularized() in statsmodels 0.8.0

### PR DESCRIPTION
**Pull request by kshedden**
Closes #3559

### CODE REVIEW

**Comment by josef-pkt** [comment]
This needs a unit test for `predict`, which is missing AFAICS.

potential bug, IIUC:
 WLS is computed with OLS on reweighted variables, but then the model attached to the results instance would be the OLS model which doesn't know about the original variable. If this is the case, then `predict` would return the weighted prediction, `wexog.dot(params)`.

**Comment by kshedden** [comment]
We're now returning the correct results type (a RegularizedResults object containing the original, not whitened endo and exog).  Tests return green on all Py3, but there seems to be a separate py2 issue (circular import?) that I can't figure out.

**Comment by josef-pkt** [comment]
there is also an extra failure, besides the import errors, in first python version on travis

```
======================================================================
ERROR: Failure: AttributeError (attribute '__doc__' of 'instancemethod' objects is not writable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/nose/loader.py", line 523, in makeTest
    return self._makeTest(obj, parent)
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/nose/loader.py", line 582, in _makeTest
    return MethodTestCase(obj)
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/nose/case.py", line 345, in __init__
    self.inst = self.cls()
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/statsmodels-0.8.0-py2.7-linux-x86_64.egg/statsmodels/base/tests/test_data.py", line 788, in __init__
    from statsmodels.genmod.generalized_linear_model import GLM
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/statsmodels-0.8.0-py2.7-linux-x86_64.egg/statsmodels/genmod/generalized_linear_model.py", line 27, in <module>
    import statsmodels.regression.linear_model as lm
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/statsmodels-0.8.0-py2.7-linux-x86_64.egg/statsmodels/regression/__init__.py", line 1, in <module>
    from .linear_model import yule_walker
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/statsmodels-0.8.0-py2.7-linux-x86_64.egg/statsmodels/regression/linear_model.py", line 1268, in <module>
    GLS.fit_regularized.__doc__ = _fit_regularized_doc
AttributeError: attribute '__doc__' of 'instancemethod' objects is not writable
```

**Comment by josef-pkt** [comment]
larger scale pep-8 changes makes the PR changeset more difficult to read

try to comment out the `__doc__` assignment to see whether it affects the travisci tests

```
GLS.fit_regularized.__doc__ = _fit_regularized_doc
OLS.fit_regularized.__doc__ = _fit_regularized_doc
WLS.fit_regularized.__doc__ = _fit_regularized_doc
```

I don't see a change in here that might cause import errors. The PR doesn't add any module level imports.

**Comment by kshedden** [comment]
OK, we're all green now.  It was the docstring issue, import errors were a red herring.  I've added unit test for predict too.

OK to merge?

**Comment by josef-pkt** [comment]
I'll give it a brief look in a few hours, but it should be ready for merge.

File: statsmodels/regression/tests/test_regression.py [hunk 1]
```diff
@@ -1175,6 +1175,27 @@ def test_regularized_refit():
...
+def test_regularized_predict():
+    n = 100
+    p = 5
+    np.random.seed(3132)
+    xmat = np.random.normal(size=(n, p))
+    yvec = xmat.sum(1) + np.random.normal(size=n)
+    wgt = np.random.uniform(1, 2, n)
+
+    for klass in WLS, GLS:
+        model1 = WLS(yvec, xmat,  weights=wgt)
```
Comment (josef-pkt) [inline statsmodels/regression/tests/test_regression.py]: WLS is hardcoded and loop variable klass is ignored

File: statsmodels/regression/linear_model.py [hunk 2]
```diff
@@ -59,6 +59,91 @@
...
+            The number of partitions desired for the distributed
+            estimation.
+        threshold : scalar or array-like
+            The threshold below which coefficients are zeroed out,
+            only used for distributed estimation
+
+        Returns
+        -------
+        An array of coefficients, or a RegressionResults object of the
+        same type returned by ``fit``.
```
Comment (josef-pkt) [inline statsmodels/regression/linear_model.py]: The current version does not return the standard results class

**Comment by kshedden** [comment]
Thanks for the review.  I think we're good to go now.  If no objections I will merge tomorrow.

**Pull request merged**